app/src/main/java/org/mozilla/fenix/compose/button/Button.kt

```diff
@@ -31,40 +31,46 @@ import org.mozilla.fenix.theme.FirefoxTheme
 import org.mozilla.fenix.theme.Theme
 
 /**
- * Primary button.
+ * Base component for buttons.
  *
  * @param text The button text to be displayed.
- * @param icon Optional button icon to be displayed.
+ * @param text [Color] to apply to the button text.
+ * @param backgroundColor The background [Color] of the button.
+ * @param icon Optional [Painter] used to display a [Icon] before the button text.
+ * @param iconTint Tint [Color] to be applied to the icon.
  * @param onClick Invoked when the user clicks on the button.
  */
 @Composable
-fun PrimaryButton(
+private fun Button(
     text: String,
+    textColor: Color,
+    backgroundColor: Color,
     icon: Painter? = null,
+    iconTint: Color,
     onClick: () -> Unit,
 ) {
-    Button(
+    androidx.compose.material.Button(
         onClick = onClick,
         modifier = Modifier
             .clip(RoundedCornerShape(size = 4.dp))
             .fillMaxWidth()
             .height(36.dp),
```

Comment:
still sad about the fixed height, but it'll require some more research to find an appropriate solution.